WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -245,11 +245,15 @@ extension PointOfSaleAggregateModel {
     func cancelThenCollectPayment() {
         Task { [weak self] in
             guard let self else { return }
-            try await cardPresentPaymentService.cancelPayment()
-            await collectCardPayment()
+            await cancelThenCollectPayment()
         }
     }
 
+    func cancelThenCollectPayment() async {
+        try? await cardPresentPaymentService.cancelPayment()
```

Comment:
Such a subtle change. I haven't internalized the dangers of [Task swallowing the error](https://forums.swift.org/t/task-initializer-with-throwing-closure-swallows-error/56066). We should be careful with this.

Reply:
Yep, it surprised me that you don't need to declare it as a throwing task or anything like that.